GithubIssueTagger/InsertionChangelogGenerator.cs

```diff
@@ -0,0 +1,212 @@
+
+            MatchCollection matches = urlRx.Matches(message);
+            foreach (Match match in matches)
+            {
+                var url = match.Value.ToLowerInvariant();
+
+                foreach (var issueRepo in issueRepository.Split(';'))
+                {
+                    var repo = issueRepo.ToLowerInvariant();
+                    if (url.Contains(repo) && url.Contains("issues"))
+                    {
+                        var regex = new Regex(@"[\D]*$");
+                        if (regex.IsMatch(url))
+                        {
+                            url = regex.Replace(url, string.Empty);
+                        }
+
+                        list.Add(url);
+                    }
+                }
+            }
+
+            return list;
+        }
+
+        public static void SaveAsHtml(IList<Commit> commits, string path)
+        {
+            if (File.Exists(path))
+            {
+                File.Delete(path);
+            }
+
+            using (var w = File.AppendText(path))
+            {
+                w.WriteLine(
+@"<!DOCTYPE html>
+<html lang=""en"" xmlns=""http://www.w3.org/1999/xhtml\"">
+<head>
+<meta charset=""utf-8"" />
+<title>Change Log</title>
+<style>
+table, th, td {
+    border: 1px solid black;
+    border-collapse: collapse;
+    padding: 15px;
+    text-align: left;
+}
+</style>
+</head>
+<body>
+<table style=""width:100%"">
+<tr>
+<th>Area</th>
+<th>Pull Request</th>
+<th>Issue(s)</th>
+<th>Commit</th>
+<th>Author</th>
+<th>Commit Message</th>
+<tr>");
+                foreach (var commit in commits)
+                {
+                    w.WriteLine(
+@$"<tr>
+<td></td>
+<td>{GetHrefOrEmpty(commit.PR)}</td>
+<td>{string.Join($"</br>", commit.Issues.Select(e => GetHrefOrEmpty(e)))}</td>
+<td><a href=""{commit.Link}"">{commit.Sha}</a></td>
+<td>{commit.Author}</ td >
+<td>{commit.Message}</ td >
+<tr> ");
+                }
+
+                w.WriteLine(
+@"</body>
+</html>");
+            }
+
+            Console.WriteLine($"Saving results file: {path}");
+
+            static string GetHrefOrEmpty(Tuple<int, string> url)
+            {
+                if (url == null)
+                {
+                    return string.Empty;
+                }
+                return $"<a href=\"{url.Item2}\">{url.Item1}</a>";
+            }
+        }
+
+        public class Commit
+        {
+            public string Sha { get; }
+            public string Author { get; }
+            public string Link { get; }
+            public string Message { get; }
+
+            public ISet<Tuple<int, string>> Issues { get; }
+
+            public Tuple<int, string> PR { get; set; }
+            
+            public Commit(string sha, string author, string link, string message)
+            {
+                Sha = sha;
+                Author = author;
+                Link = link;
+                Message = message?
+                            .Replace(",", " ")
```

Comment:
why remove `,`?

Reply:
I don't remember tbh. I'll remove it and see if anything breaks :D